{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "CONTRIBUTOR",
  "body": "<!--\r\nSTOP -- PLEASE READ!\r\n\r\nGitHub is not the right place for support requests.\r\n\r\nIf you're looking for help, check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).\r\n\r\nYou can also post your question on the [Kubernetes Slack](http://slack.k8s.io/) or the [Discuss Kubernetes](https://discuss.kubernetes.io/) forum.\r\n\r\nIf the matter is security related, please disclose it privately via https://kubernetes.io/security/.\r\n-->\r\n\r\nIn the code below, kubelet rejects pods with QOS type guaranteed when under disk pressure, but when kubelet is **also** under memory pressure, those pods are accepted, is this behavior expected? Why is that?\r\n\r\nhttps://github.com/kubernetes/kubernetes/blob/2896fb0be8515e909df71bf29074acf32c54c969/pkg/kubelet/eviction/eviction_manager.go#L128-L164\r\n",
  "closed_at": "2019-11-04T21:44:25Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 4,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84403/comments",
  "created_at": "2019-10-26T14:46:47Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84403/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/84403",
  "id": 512843992,
  "labels": [
    {
      "color": "0052cc",
      "default": false,
      "description": null,
      "id": 116719829,
      "name": "area/kubelet",
      "node_id": "MDU6TGFiZWwxMTY3MTk4Mjk=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/area/kubelet"
    },
    {
      "color": "d455d0",
      "default": false,
      "description": "Categorizes issue or PR as a support question.",
      "id": 130318610,
      "name": "kind/support",
      "node_id": "MDU6TGFiZWwxMzAzMTg2MTA=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/support"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Node.",
      "id": 173493665,
      "name": "sig/node",
      "node_id": "MDU6TGFiZWwxNzM0OTM2NjU=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/node"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84403/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWU1MTI4NDM5OTI=",
  "number": 84403,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "why kubelet admits guaranteed pods when under disk pressure?",
  "updated_at": "2019-11-04T21:44:25Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/84403",
  "user": {
    "avatar_url": "https://avatars1.githubusercontent.com/u/23180738?v=4",
    "events_url": "https://api.github.com/users/langyenan/events{/privacy}",
    "followers_url": "https://api.github.com/users/langyenan/followers",
    "following_url": "https://api.github.com/users/langyenan/following{/other_user}",
    "gists_url": "https://api.github.com/users/langyenan/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/langyenan",
    "id": 23180738,
    "login": "langyenan",
    "node_id": "MDQ6VXNlcjIzMTgwNzM4",
    "organizations_url": "https://api.github.com/users/langyenan/orgs",
    "received_events_url": "https://api.github.com/users/langyenan/received_events",
    "repos_url": "https://api.github.com/users/langyenan/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/langyenan/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/langyenan/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/langyenan"
  }
}